api/control/v1alpha1/types.go

```diff
@@ -490,8 +490,24 @@ type ReturnedState struct {
 type PropagationData struct {
 	// `lastReturnedUpdateTimestamp` is the time of the last update to any
 	// of the returned object state in the core.
-	// Before the first such update, this is the zero value of `time.Time`.
+	// Before the first such update, this holds the zero value of `time.Time`.
 	LastReturnedUpdateTimestamp metav1.Time `json:"lastReturnedUpdateTimestamp"`
+
+	// `lastGeneration` is that last `ObjectMeta.Generation` from the WDS that
+	// propagated to the WEC. This is not to imply that it was successfully applied there;
+	// for that, see `lastGenerationIsApplied`.
+	// Zero means that none has yet propagated there.
+	LastGeneration int64 `json:"lastGeneration"`
+
+	// `lastGenerationIsApplied` indicates whether `lastGeneration` has been successfully
+	// applied in the WEC. This is `false` while no attempt has yet been made.
+	LastGenerationIsApplied bool `json:"lastGenerationIsApplied"`
+
+	// `lastCurrencyUpdateTime` is the time of the latest update to either
+	// `lastGeneration` or `lastGenerationIsApplied`. More precisely, it is
+	// the time when the core became informed of the update.
+	// Before the first such update, this holds the zero value of `time.Time`.
+	LastCurrencyUpdateTime metav1.Time `json:"lastCurrencyUpdateTime"`
```

Comment:
Perhaps due to not being a native speaker, this name does not tell me much. With this timestamp and the `LastGenerationIsApplied` bit one can infer whether this is a `LastGenerationUpdateTimestamp` or a `LastGenerationApplicationTimestamp`. The current name/field-documentation do not explicitly imply this usefulness.

Reply:
@vMaroon: I am not sure I understand your point. Did you omit a word, meaning to say "can not infer" rather than "can infer"? The second sentence of the comment is intended to clarify that the time value in this field is read from a clock in the core in every case. Do you find that unclear?

Indeed I see no reason to expect `LastGenerationIsApplied` to change from `true` to `false` without _also_ there being a change to `LastGeneration`. I have not attempted to optimize the API in light of that exclusion, nor am I sure that we _should_ build that exclusion into the API.

Reply:
The point is I have no idea what this name is attempting to say. If there are alternatives, I think they are preferable.

`LastGenerationIsApplied` bit sets the meaning of this timestamp. Perhaps this can be reflected through the name/documentation.

Reply:
In every case this is a timestamp read from a clock in the core. More particularly, this is `time.Now()` from the status controller, executed near the time it updates either `lastGeneration` or `lastGenerationIsApplied`.